daemon/graphdriver/overlay/overlay.go

```diff
@@ -380,12 +377,12 @@ func (d *Driver) Put(id string) error {
 		return nil
 	}
 
-	mount.count--
-	if mount.count > 0 {
+	count := d.active.Decr(id)
+	if count > 0 {
 		return nil
 	}
-
-	defer delete(d.active, id)
+	mount := m.(*ActiveMount)
+	defer d.active.Delete(id)
```

Comment:
Why not let the `refcount.Counter` auto-delete once the count == 0?
After all, with this you can have a race condition where you call `Decr()`, get the count, something else increases it, but then you delete it because when you checked, it wasn't in use.